shared/haml/user_header.haml

```diff
@@ -107,6 +107,17 @@
       }
     }
 
+    var displayNameSpan = document.querySelector('.header_button.header_user.user_menu .display_name');
+    if (displayNameSpan) {
+      var dataName = displayNameSpan.dataset.shortname;
+      var id = displayNameSpan.dataset.id;
+      var cookieMatch = document.cookie.match('(^|;) ?' + cookieKey + '=([^;]*)(;|$)');
+      var storedName = cookieMatch && cookieMatch[2];
+      if (storedName && !id) {
+        displayNameSpan.innerHTML = displayNameSpan.innerHTML.replace(dataName, storedName);
+      }
+    }
```

Comment:
it feels like we could factor out some code between this block and the one above it?

Reply:
I extracted the two shared lines. Is that what you were thinking?

Reply:
Just by glancing, it looks as though there is a bigger possible duplication to factor out; we seem to have basically the same chunk of code which could be parameterised with `user_name` vs. `display_name` and `nameSpan` vs. `displayNameSpan`.